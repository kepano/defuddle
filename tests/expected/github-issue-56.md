Example repo here: [https://github.com/jmorrell/defuddle-cloudflare-example](https://github.com/jmorrell/defuddle-cloudflare-example)

I was looking forward to ditching my tenuous `readbilityjs` fork in my workers project. Defuddle ultimately does run to completion, but I ran into a couple of issues. Supporting this environment fully is likely challenging since `JSDOM` does not work within the Worker environment. I suspect the same will be true of the deno and bun runtimes.

I was able to work around this by using the browser version along with [linkedom](https://github.com/WebReflection/linkedom), however this is not an exact replacement for JSDOM and doesn't implement all the CSS functionality.

Since defuddle relies on these style heuristics, **I'm not sure there is a great path to supporting the full functionality in this environment**, but I wanted to open an issue for discussion and to document these issues for anyone else who might hit this.

When running I get two errors:

```
Defuddle: Error evaluating media queries: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```

This is due to linkedom not implementing `doc.styleSheets`: [https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213](https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213)

This could be silenced by falling back to `[]` if `doc.styleSheets` isn't present, however that may not be the desired behavior.

```
const sheets = Array.from(doc.styleSheets ?? [])
```

The second issue is due to `getComputedStyle` not being supported by linkedom.

```
Defuddle Error processing document: TypeError: e3.getComputedStyle is not a function
```

If you feel like there's nothing to do, or supporting Workers is out-of-scope for the project, feel free to close the issue