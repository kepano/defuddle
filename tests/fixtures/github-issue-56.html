<!DOCTYPE html>
<html
  lang="en"
  
  data-color-mode="auto" data-light-theme="light" data-dark-theme="dark"
  data-a11y-animated-images="system" data-a11y-link-underlines="true"
  
  >



  <head>
    <meta charset="utf-8">
  <link rel="dns-prefetch" href="https://github.githubassets.com">
  <link rel="dns-prefetch" href="https://avatars.githubusercontent.com">
  <link rel="dns-prefetch" href="https://github-cloud.s3.amazonaws.com">
  <link rel="dns-prefetch" href="https://user-images.githubusercontent.com/">
  <link rel="preconnect" href="https://github.githubassets.com" crossorigin>
  <link rel="preconnect" href="https://avatars.githubusercontent.com">

  

  <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/light-d1334f2b22bf.css" /><link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/light_high_contrast-f695a361c6b2.css" /><link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/dark-f73a069fd33e.css" /><link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/dark_high_contrast-3a0d87f72ad4.css" /><link data-color-theme="light" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/light-d1334f2b22bf.css" /><link data-color-theme="light_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/light_high_contrast-f695a361c6b2.css" /><link data-color-theme="light_colorblind" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/light_colorblind-367eb9a4565a.css" /><link data-color-theme="light_colorblind_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/light_colorblind_high_contrast-183adc0db479.css" /><link data-color-theme="light_tritanopia" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/light_tritanopia-2ddc677c041d.css" /><link data-color-theme="light_tritanopia_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/light_tritanopia_high_contrast-649962a5702a.css" /><link data-color-theme="dark" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark-f73a069fd33e.css" /><link data-color-theme="dark_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_high_contrast-3a0d87f72ad4.css" /><link data-color-theme="dark_colorblind" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_colorblind-b17a8392e6c4.css" /><link data-color-theme="dark_colorblind_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_colorblind_high_contrast-e9ff47cedc2b.css" /><link data-color-theme="dark_tritanopia" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_tritanopia-a1cc7dba9f73.css" /><link data-color-theme="dark_tritanopia_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_tritanopia_high_contrast-6c4dd39e2b0f.css" /><link data-color-theme="dark_dimmed" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_dimmed-55459b36aa6d.css" /><link data-color-theme="dark_dimmed_high_contrast" crossorigin="anonymous" media="all" rel="stylesheet" data-href="https://github.githubassets.com/assets/dark_dimmed_high_contrast-9a0ef6e40ed3.css" />


    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/primer-primitives-dc7ca6859caf.css" />
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/primer-03a65c451725.css" />
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/global-d1aed5b4075c.css" />
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/github-db7b386c190b.css" />
  <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://github.githubassets.com/assets/repository-b58e401b73ae.css" />

  


  <script type="application/json" id="client-env">{"locale":"en","featureFlags":["alternate_user_config_repo","api_insights_show_missing_data_banner","attestations_filtering","attestations_sorting","client_version_header","code_scanning_security_configuration_ternary_state","codespaces_prebuild_region_target_update","contact_requests_implicit_opt_in","contentful_lp_copilot_extensions","contentful_lp_flex_features","contentful_lp_footnotes","copilot_activity_report","copilot_chat_attach_multiple_images","copilot_chat_floating_sidebar_focus_trap","copilot_chat_group_notifications","copilot_chat_navigable_refs","copilot_chat_vision_in_claude","copilot_chat_vision_skip_thread_create","copilot_chat_wholearea_dd","copilot_custom_copilots_feature_preview","copilot_custom_copilots_images","copilot_duplicate_thread","copilot_free_to_paid_telem","copilot_ftp_hyperspace_upgrade_prompt","copilot_ftp_settings_upgrade","copilot_ftp_upgrade_to_pro_from_models","copilot_ftp_your_copilot_settings","copilot_immersive_draft_issue_template_form_ui","copilot_immersive_structured_model_picker","copilot_no_floating_button","copilot_read_shared_conversation","copilot_spaces_input_menu_select","copilot_spark_allow_empty_commit","copilot_spark_single_user_iteration","copilot_spark_use_billing_headers","copilot_task_oriented_assistive_prompts","copilot_workbench_connection_reload_banner","copilot_workbench_iterate_panel","copilot_workbench_preview_analytics","copilot_workbench_refresh_on_wsod","custom_copilots_128k_window","custom_copilots_capi_mode","custom_copilots_file_uploads","direct_to_salesforce","dotcom_chat_client_side_skills","failbot_report_error_react_apps_on_page","ghost_pilot_confidence_truncation_25","ghost_pilot_confidence_truncation_40","insert_before_patch","issue_fields_report_usage","issues_catch_non_json_graphql_response","issues_comment_load_user_settings","issues_label_search_url","issues_preserve_tokens_in_urls","issues_react_blur_item_picker_on_close","issues_react_bots_timeline_pagination","issues_react_create_milestone","issues_react_optimistic_markdown","issues_react_prohibit_title_fallback","issues_react_remove_placeholders","issues_template_picker_redirect","lifecycle_label_name_updates","link_contact_sales_swp_marketo","marketing_pages_search_explore_provider","memex_mwl_filter_field_delimiter","nonreporting_relay_graphql_status_codes","primer_react_select_panel_with_modern_action_list","remove_child_patch","sample_network_conn_type","scheduled_reminders_updated_limits","site_homepage_contentful","site_msbuild_hide_integrations","site_msbuild_launch","site_msbuild_webgl_hero","spark_commit_on_default_branch","spark_sync_repository_after_iteration","swp_enterprise_contact_form","use_paginated_repo_picker_cost_center_form","viewscreen_sandbox","workbench_default_sonnet4","workbench_store_readonly"],"copilotApiUrl":"https://api.individual.githubcopilot.com"}</script>

  <title>Defuddle on Cloudflare Workers · Issue #56 · kepano/defuddle · GitHub</title>

  <meta name="expected-hostname" content="github.com">

  <meta name="description" content="Example repo here: https://github.com/jmorrell/defuddle-cloudflare-example I was looking forward to ditching my tenuous readbilityjs fork in my workers project. Defuddle ultimately does run to completion, but I ran into a couple of issue...">

  <script type="application/ld+json">{"@context":"https://schema.org","@type":"DiscussionForumPosting","headline":"Defuddle on Cloudflare Workers","articleBody":"Example repo here: https://github.com/jmorrell/defuddle-cloudflare-example\n\nI was looking forward to ditching my tenuous `readbilityjs` fork in my workers project. Defuddle ultimately does run to completion, but I ran into a couple of issues. Supporting this environment fully is likely challenging since `JSDOM` does not work within the Worker environment. I suspect the same will be true of the deno and bun runtimes.\n\nI was able to work around this by using the browser version along with [linkedom](https://github.com/WebReflection/linkedom), however this is not an exact replacement for JSDOM and doesn't implement all the CSS functionality.\n\nSince defuddle relies on these style heuristics, **I'm not sure there is a great path to supporting the full functionality in this environment**, but I wanted to open an issue for discussion and to document these issues for anyone else who might hit this.\n\nWhen running I get two errors:\n\n```\nDefuddle: Error evaluating media queries: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))\n```\n\nThis is due to linkedom not implementing `doc.styleSheets`: https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213\n\nThis could be silenced by falling back to `[]` if `doc.styleSheets` isn't present, however that may not be the desired behavior.\n\n```ts\nconst sheets = Array.from(doc.styleSheets ?? [])\n```\n\nThe second issue is due to `getComputedStyle` not being supported by linkedom.\n\n```\nDefuddle Error processing document: TypeError: e3.getComputedStyle is not a function\n```\n\nIf you feel like there's nothing to do, or supporting Workers is out-of-scope for the project, feel free to close the issue","author":{"url":"https://github.com/jmorrell","@type":"Person","name":"jmorrell"},"datePublished":"2025-05-25T20:35:48.000Z","interactionStatistic":{"@type":"InteractionCounter","interactionType":"https://schema.org/CommentAction","userInteractionCount":6},"url":"https://github.com/56/defuddle/issues/56"}</script> 

  </head>

  <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
    <div data-turbo-body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
      <div class="position-relative header-wrapper js-header-wrapper ">
        <a href="#start-of-content" data-skip-target-assigned="false" class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content">Skip to content</a>
        <header class="HeaderMktg header-logged-out js-details-container js-header Details f4 py-3" role="banner" data-is-top="true" data-color-mode=light data-light-theme=light data-dark-theme=dark>
          <h2 class="sr-only">Navigation Menu</h2>
          <div class="d-flex flex-column flex-lg-row flex-items-center px-3 px-md-4 px-lg-5 height-full position-relative z-1">
            <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
              <a class="mr-lg-3 color-fg-inherit flex-order-2 js-prevent-focus-on-mobile-nav" href="/" aria-label="Homepage">
                <svg height="32" aria-hidden="true" viewBox="0 0 24 24" version="1.1" width="32" data-view-component="true" class="octicon octicon-mark-github">
                  <path d="M12 1C5.923 1 1 5.923 1 12c0 4.867 3.149 8.979 7.521 10.436.55.096.756-.233.756-.522 0-.262-.013-1.128-.013-2.049-2.764.509-3.479-.674-3.699-1.292-.124-.317-.66-1.293-1.127-1.554-.385-.207-.936-.715-.014-.729.866-.014 1.485.797 1.691 1.128.99 1.663 2.571 1.196 3.204.907.096-.715.385-1.196.701-1.471-2.448-.275-5.005-1.224-5.005-5.432 0-1.196.426-2.186 1.128-2.956-.111-.275-.496-1.402.11-2.915 0 0 .921-.288 3.024 1.128a10.193 10.193 0 0 1 2.75-.371c.936 0 1.871.123 2.75.371 2.104-1.43 3.025-1.128 3.025-1.128.605 1.513.221 2.64.111 2.915.701.77 1.127 1.747 1.127 2.956 0 4.222-2.571 5.157-5.019 5.432.399.344.743 1.004.743 2.035 0 1.471-.014 2.654-.014 3.025 0 .289.206.632.756.522C19.851 20.979 23 16.854 23 12c0-6.077-4.922-11-11-11Z"></path>
                </svg>
              </a>
            </div>
          </div>
        </header>
      </div>

      <div id="start-of-content" class="show-on-focus"></div>

      <div id="js-flash-container" data-turbo-cache="false"></div>

      <main id="js-repo-pjax-container">
        <div class="pt-3">
          <div class="container-xl clearfix new-discussion-timeline px-3 px-md-4 px-lg-5">
            <div class="repository-content">
              <div class="js-navigation-container" data-pjax-container>
                <div class="d-flex">
                  <div class="flex-auto min-width-0 width-fit mr-3">
                    <div class="gh-header-show">
                      <div class="gh-header-title">
                        <h1 class="gh-header-title" data-testid="issue-title">
                          <span class="js-issue-title markdown-title" data-testid="issue-title-text">Defuddle on Cloudflare Workers</span>
                          <span class="gh-header-number">#56</span>
                        </h1>
                      </div>
                      <div class="gh-header-meta">
                        <div class="gh-header-meta-item">
                          <span class="State State--open" data-testid="issue-state-badge">
                            <svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-issue-opened">
                              <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"></path>
                              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"></path>
                            </svg>
                            Open
                          </span>
                        </div>
                        <div class="gh-header-meta-item">
                          <a class="author text-bold Link--muted" data-hovercard-type="user" data-hovercard-url="/users/jmorrell/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/jmorrell">jmorrell</a>
                          opened this issue on
                          <relative-time datetime="2025-05-25T20:35:48Z" class="no-wrap" title="May 25, 2025, 8:35 PM GMT">May 25, 2025</relative-time>
                          ·
                          6 comments
                        </div>
                      </div>
                    </div>

                    <div class="timeline-comment-wrapper js-comment-container">
                      <div class="timeline-comment" id="issue-3089558116">
                        <div class="timeline-comment-header">
                          <div class="timeline-comment-header-text f5 text-normal">
                            <strong>
                              <a class="author" data-hovercard-type="user" data-hovercard-url="/users/jmorrell/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/jmorrell">jmorrell</a>
                            </strong>
                            commented
                            <relative-time datetime="2025-05-25T20:35:48Z" class="no-wrap" title="May 25, 2025, 8:35 PM GMT">on May 25, 2025</relative-time>
                          </div>
                        </div>

                        <div class="comment-body markdown-body js-comment-body">
                          <p>Example repo here: <a href="https://github.com/jmorrell/defuddle-cloudflare-example">https://github.com/jmorrell/defuddle-cloudflare-example</a></p>
                          <p>I was looking forward to ditching my tenuous <code>readbilityjs</code> fork in my workers project. Defuddle ultimately does run to completion, but I ran into a couple of issues. Supporting this environment fully is likely challenging since <code>JSDOM</code> does not work within the Worker environment. I suspect the same will be true of the deno and bun runtimes.</p>
                          <p>I was able to work around this by using the browser version along with <a href="https://github.com/WebReflection/linkedom">linkedom</a>, however this is not an exact replacement for JSDOM and doesn't implement all the CSS functionality.</p>
                          <p>Since defuddle relies on these style heuristics, <strong>I'm not sure there is a great path to supporting the full functionality in this environment</strong>, but I wanted to open an issue for discussion and to document these issues for anyone else who might hit this.</p>
                          <p>When running I get two errors:</p>
                          <div class="highlight highlight-text-plain notranslate position-relative overflow-auto">
                            <pre>Defuddle: Error evaluating media queries: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))</pre>
                          </div>
                          <p>This is due to linkedom not implementing <code>doc.styleSheets</code>: <a href="https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213">https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213</a></p>
                          <p>This could be silenced by falling back to <code>[]</code> if <code>doc.styleSheets</code> isn't present, however that may not be the desired behavior.</p>
                          <div class="highlight highlight-source-ts notranslate position-relative overflow-auto">
                            <pre><span class="pl-k">const</span> <span class="pl-s1">sheets</span> <span class="pl-c1">=</span> <span class="pl-v">Array</span><span class="pl-kos">.</span><span class="pl-en">from</span><span class="pl-kos">(</span><span class="pl-s1">doc</span><span class="pl-kos">.</span><span class="pl-c1">styleSheets</span> <span class="pl-c1">??</span> <span class="pl-kos">[</span><span class="pl-kos">]</span><span class="pl-kos">)</span></pre>
                          </div>
                          <p>The second issue is due to <code>getComputedStyle</code> not being supported by linkedom.</p>
                          <div class="highlight highlight-text-plain notranslate position-relative overflow-auto">
                            <pre>Defuddle Error processing document: TypeError: e3.getComputedStyle is not a function</pre>
                          </div>
                          <p>If you feel like there's nothing to do, or supporting Workers is out-of-scope for the project, feel free to close the issue</p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
