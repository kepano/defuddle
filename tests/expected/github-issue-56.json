{
  "metadata": {
    "title": "Defuddle on Cloudflare Workers",
    "author": "jmorrell",
    "site": "jmorrell",
    "published": "2025-05-25T20:35:48.000Z"
  },
  "content": {
    "html": "<div><p>Example repo here: <a href=\"https://github.com/jmorrell/defuddle-cloudflare-example\">https://github.com/jmorrell/defuddle-cloudflare-example</a></p> <p>I was looking forward to ditching my tenuous <code>readbilityjs</code> fork in my workers project. Defuddle ultimately does run to completion, but I ran into a couple of issues. Supporting this environment fully is likely challenging since <code>JSDOM</code> does not work within the Worker environment. I suspect the same will be true of the deno and bun runtimes.</p><p>I was able to work around this by using the browser version along with <a href=\"https://github.com/WebReflection/linkedom\">linkedom</a>, however this is not an exact replacement for JSDOM and doesn't implement all the CSS functionality.</p><p>Since defuddle relies on these style heuristics, <strong>I'm not sure there is a great path to supporting the full functionality in this environment</strong>, but I wanted to open an issue for discussion and to document these issues for anyone else who might hit this.</p><p>When running I get two errors:</p><pre><code>Defuddle: Error evaluating media queries: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))</code></pre> <p>This is due to linkedom not implementing <code>doc.styleSheets</code>: <a href=\"https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213\">https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213</a></p> <p>This could be silenced by falling back to <code>[]</code> if <code>doc.styleSheets</code> isn't present, however that may not be the desired behavior.</p><pre><code>const sheets = Array.from(doc.styleSheets ?? [])</code></pre> <p>The second issue is due to <code>getComputedStyle</code> not being supported by linkedom.</p><pre><code>Defuddle Error processing document: TypeError: e3.getComputedStyle is not a function</code></pre> <p>If you feel like there's nothing to do, or supporting Workers is out-of-scope for the project, feel free to close the issue</p></div>",
    "markdown": "Example repo here: [https://github.com/jmorrell/defuddle-cloudflare-example](https://github.com/jmorrell/defuddle-cloudflare-example)\n\nI was looking forward to ditching my tenuous `readbilityjs` fork in my workers project. Defuddle ultimately does run to completion, but I ran into a couple of issues. Supporting this environment fully is likely challenging since `JSDOM` does not work within the Worker environment. I suspect the same will be true of the deno and bun runtimes.\n\nI was able to work around this by using the browser version along with [linkedom](https://github.com/WebReflection/linkedom), however this is not an exact replacement for JSDOM and doesn't implement all the CSS functionality.\n\nSince defuddle relies on these style heuristics, **I'm not sure there is a great path to supporting the full functionality in this environment**, but I wanted to open an issue for discussion and to document these issues for anyone else who might hit this.\n\nWhen running I get two errors:\n\n```\nDefuddle: Error evaluating media queries: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))\n```\n\nThis is due to linkedom not implementing `doc.styleSheets`: [https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213](https://github.com/kepano/defuddle/blob/cb4291db0f24cac0d0674d9e35fc0089338da2da/src/defuddle.ts#L213)\n\nThis could be silenced by falling back to `[]` if `doc.styleSheets` isn't present, however that may not be the desired behavior.\n\n```\nconst sheets = Array.from(doc.styleSheets ?? [])\n```\n\nThe second issue is due to `getComputedStyle` not being supported by linkedom.\n\n```\nDefuddle Error processing document: TypeError: e3.getComputedStyle is not a function\n```\n\nIf you feel like there's nothing to do, or supporting Workers is out-of-scope for the project, feel free to close the issue"
  }
}